dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY HTML INODE=" + id.getId());
+
+        Template template;
+        Template hostVariablesTemplate = null;
+
+        // creates the context where to place the variables
+        response.setContentType(CHARSET);
+        request.setAttribute("EDIT_MODE", Boolean.TRUE);
+        Context context = VelocityUtil.getWebContext(request, response);
+
+        IHTMLPage htmlPage = VelocityUtil.getPage(id, request, false, context);
+
+        context.put("dotPageContent", new ContentMap(((Contentlet) htmlPage), user, mode, host, context));
+
+
+        com.dotmarketing.portlets.templates.model.Template cmsTemplate = APILocator.getHTMLPageAssetAPI()
+            .getTemplate(htmlPage, true);
+
+        if (cmsTemplate == null) {// DOTCMS-4051
+            cmsTemplate = new com.dotmarketing.portlets.templates.model.Template();
+            Logger.debug(VelocityPreviewServlet.class, "HTMLPAGE TEMPLATE NOT FOUND");
+        }
+
+        Identifier templateIdentifier = APILocator.getIdentifierAPI()
+            .find(cmsTemplate);
+
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY TEMPLATE INODE=" + cmsTemplate.getInode());
+
+        VelocityUtil.makeBackendContext(context, htmlPage, cmsTemplate.getInode(), id.getURI(), request, mode, host);
+
+
+        PageContextBuilder builder = new PageContextBuilder(htmlPage, user, PageMode.PREVIEW_MODE);
+        builder.addAll(context);
+
+        hostVariablesTemplate = VelocityUtil.getEngine()
+            .getTemplate(PageMode.EDIT_MODE.name() + File.separator + host.getIdentifier() + "." + VelocityType.SITE.fileExtension);
+
+        if (cmsTemplate.isDrawed()) {// We have a designed template
+            // Setting some theme variables
+            Map<String, Object> dotThemeData = DotTemplateTool.theme(cmsTemplate.getTheme(), host.getIdentifier());
+            context.put("dotTheme", dotThemeData);
+            context.put("dotThemeLayout", DotTemplateTool.themeLayout(cmsTemplate.getInode()));
+            // Our designed template
+            template = VelocityUtil.getEngine()
+                .getTemplate((String) dotThemeData.get("templatePath"));
+        } else {
+            template = VelocityUtil.getEngine()
+                .getTemplate(PageMode.EDIT_MODE.name() + File.separator + templateIdentifier.getId() + "."
+                        + VelocityType.TEMPLATE.fileExtension);
+        }
+
+        PrintWriter out = response.getWriter();
+        request.setAttribute("velocityContext", context);
+        try {
+            if (builder.getWidgetPreExecute() != null) {
+                VelocityUtil.getEngine()
+                    .evaluate(context, out, "", builder.getWidgetPreExecute());
+            }
+            if (hostVariablesTemplate != null)
+                hostVariablesTemplate.merge(context, out);
+            template.merge(context, out);
+
+        } catch (ParseErrorException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)